app/src/main/java/org/mozilla/fenix/components/metrics/MetricController.kt

```diff
@@ -93,13 +95,33 @@ internal class ReleaseMetricController(
     init {
         Facts.registerProcessor(object : FactProcessor {
             override fun process(fact: Fact) {
-                fact.toEvent()?.also {
-                    track(it)
-                }
+                fact.process()
             }
         })
     }
 
+    @VisibleForTesting
+    internal fun Fact.process(): Unit = when (component to item) {
+        Component.FEATURE_PROMPTS to LoginDialogFacts.Items.DISPLAY -> {
+            LoginDialog.displayed.record(NoExtras())
+        }
+        Component.FEATURE_PROMPTS to LoginDialogFacts.Items.CANCEL -> {
+            LoginDialog.cancelled.record(NoExtras())
+        }
+        Component.FEATURE_PROMPTS to LoginDialogFacts.Items.NEVER_SAVE -> {
+            LoginDialog.neverSave.record(NoExtras())
+        }
+        Component.FEATURE_PROMPTS to LoginDialogFacts.Items.SAVE -> {
+            LoginDialog.saved.record(NoExtras())
+        }
+        else -> {
```

Comment:
If we have a Fact that is not expected, should we call track(it)?  Or log it or throw an exception?